### DISCUSSION TITLE
docs/micro-service

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#120] [General]
# docs/micro-service

GoFrame框架的微服务开发特性，包括支持HTTP和GRPC协议的微服务组件、低藕性设计、服务注册发现组件的使用以及链路跟踪能力的示例教程，帮助团队快速实现微服务的开发与转型。示例涵盖了如何使用GoFrame进行微服务架构的设置与通信。

https://goframe.org.cn/docs/micro-service

**Comment by apefuu** [comment]
我想问一下，大仓模式下，几个不同的微服务需要有公共的依赖代码，但我又不想df init app/xxx -a 出来新的微服务通过rpc去调用，单纯只是依赖引入就好，需要怎么组织代码结构呢

**Reply by gqcn** [reply]
@apefuu 
- 在大仓项目模板的最外层有一个`utility`目录，用于存放所有微服务都可以引用的公共组件，在这里 https://github.com/gogf/template-mono/tree/main/utility 。
- 如果是特定的微服务可内聚封装的组件，但需要对其他服务代码引用，就放到特定的微服务目录下的`utility`目录维护，例如在这里创建一个公共目录即可 https://github.com/gogf/template-mono/tree/main/app/svc-template

**Comment by 1378496782** [comment]
client := g.Client()
	client.SetDiscovery(gsvc.GetRegistry())
